website/pages/en/cookbook/how-to-secure-api-keys-using-nextjs-server-components.mdx

```diff
@@ -0,0 +1,123 @@
+## How to Secure API Keys Using Next.js Server Components
+
+### Overview
+
+We can use [NextJS server components](https://nextjs.org/docs/app/building-your-application/rendering/server-components) to properly secure our API key from exposure in the front-end of our dapp. To further increase our API key security, we can also [restrict our API key to certain subgraphs or domains in Subgraph Studio](https://thegraph.com/docs/en/cookbook/upgrading-a-subgraph/#securing-your-api-key).
```

Comment:
```suggestion
We can use [NextJS server components](https://nextjs.org/docs/app/building-your-application/rendering/server-components) to properly secure our API key from exposure in the front-end of our dapp. To further increase our API key security, we can also [restrict our API key to certain subgraphs or domains in Subgraph Studio](/cookbook/upgrading-a-subgraph/#securing-your-api-key).
```
Prefer root-relative URLs (minus the locale) over absolute ones so navigating on staging or local dev doesn't direct you to production without you noticing. 😛